File: platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/maps/NativeMapOptions.java
package org.maplibre.android.maps;

public class NativeMapOptions {

  private final float pixelRatio;
  private final boolean crossSourceCollisions;

  private final boolean actionJournalEnabled;
  private final String actionJournalPath;
  private final long actionJournalLogFileSize;
  private final long actionJournalLogFileCount;
  private final int actionJournalRenderingReportInterval;
  private final boolean asyncRendererCleanup;

  private final boolean enableFastPFOR;
  public NativeMapOptions(MapLibreMapOptions options) {
    pixelRatio = options.getPixelRatio();
    crossSourceCollisions = options.getCrossSourceCollisions();

    actionJournalEnabled = options.getActionJournalEnabled();
    actionJournalPath = options.getActionJournalPath();
    actionJournalLogFileSize = options.getActionJournalLogFileSize();
    actionJournalLogFileCount = options.getActionJournalLogFileCount();
    actionJournalRenderingReportInterval = options.getActionJournalRenderingReportInterval();
    asyncRendererCleanup = options.getAsyncRendererCleanup();
    enableFastPFOR = options.getEnableFastPFOR();
  }

  public NativeMapOptions(float pixelRatio, boolean crossSourceCollisions) {
    this.pixelRatio = pixelRatio;
    this.crossSourceCollisions = crossSourceCollisions;

    actionJournalEnabled = false;
    actionJournalPath = "";
    actionJournalLogFileSize = 0;
    actionJournalLogFileCount = 0;
    actionJournalRenderingReportInterval = 0;

    asyncRendererCleanup = false;
    enableFastPFOR = false;
  }

  public float pixelRatio() {
    return pixelRatio;
  }

  public boolean crossSourceCollisions() {
    return crossSourceCollisions;
  }

  public boolean actionJournalEnabled() {
    return actionJournalEnabled;
  }

  public String actionJournalPath() {
    return actionJournalPath;
  }

  public long actionJournalLogFileSize() {
    return actionJournalLogFileSize;
  }

  public long actionJournalLogFileCount() {
    return actionJournalLogFileCount;
  }

  public int actionJournalRenderingReportInterval() {
    return actionJournalRenderingReportInterval;
  }

  public boolean asyncRendererCleanup() {
    return asyncRendererCleanup;
  }

  public boolean enableFastPFOR() {
    return enableFastPFOR;
  }
}

File: platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/maps/NativeMapOptions.kt
package org.maplibre.android.maps

/**
 * The subset of [MapLibreMapOptions] that is passed to the native map.
 *
 * The fields of this class are read from the native peer, they must not be renamed.
 */
class NativeMapOptions {
    private val pixelRatio: Float
    private val crossSourceCollisions: Boolean

    private val actionJournalEnabled: Boolean
    private val actionJournalPath: String
    private val actionJournalLogFileSize: Long
    private val actionJournalLogFileCount: Long
    private val actionJournalRenderingReportInterval: Int
    private val asyncRendererCleanup: Boolean

    private val fastPFOREnabled: Boolean

    constructor(options: MapLibreMapOptions) {
        pixelRatio = options.pixelRatio
        crossSourceCollisions = options.crossSourceCollisions

        actionJournalEnabled = options.actionJournalEnabled
        actionJournalPath = options.actionJournalPath
        actionJournalLogFileSize = options.actionJournalLogFileSize
        actionJournalLogFileCount = options.actionJournalLogFileCount
        actionJournalRenderingReportInterval = options.actionJournalRenderingReportInterval
        asyncRendererCleanup = options.asyncRendererCleanup
        fastPFOREnabled = options.fastPFOREnabled
    }

    constructor(pixelRatio: Float, crossSourceCollisions: Boolean) {
        this.pixelRatio = pixelRatio
        this.crossSourceCollisions = crossSourceCollisions

        actionJournalEnabled = false
        actionJournalPath = ""
        actionJournalLogFileSize = 0
        actionJournalLogFileCount = 0
        actionJournalRenderingReportInterval = 0

        asyncRendererCleanup = false
        fastPFOREnabled = false
    }

    fun pixelRatio(): Float = pixelRatio

    fun crossSourceCollisions(): Boolean = crossSourceCollisions

    fun actionJournalEnabled(): Boolean = actionJournalEnabled

    fun actionJournalPath(): String = actionJournalPath

    fun actionJournalLogFileSize(): Long = actionJournalLogFileSize

    fun actionJournalLogFileCount(): Long = actionJournalLogFileCount

    fun actionJournalRenderingReportInterval(): Int = actionJournalRenderingReportInterval

    fun asyncRendererCleanup(): Boolean = asyncRendererCleanup

    fun fastPFOREnabled(): Boolean = fastPFOREnabled
}
